merlin-framework/src/main/java/gov/nasa/jpl/aerie/merlin/framework/ReplayingTask.java

```diff
@@ -1,71 +1,227 @@
 package gov.nasa.jpl.aerie.merlin.framework;
 
+import gov.nasa.jpl.aerie.merlin.protocol.driver.CellId;
 import gov.nasa.jpl.aerie.merlin.protocol.driver.Scheduler;
+import gov.nasa.jpl.aerie.merlin.protocol.driver.Topic;
+import gov.nasa.jpl.aerie.merlin.protocol.model.CellType;
+import gov.nasa.jpl.aerie.merlin.protocol.model.Condition;
 import gov.nasa.jpl.aerie.merlin.protocol.model.Task;
 import gov.nasa.jpl.aerie.merlin.protocol.model.TaskFactory;
 import gov.nasa.jpl.aerie.merlin.protocol.types.Duration;
 import gov.nasa.jpl.aerie.merlin.protocol.types.TaskStatus;
-import org.apache.commons.lang3.mutable.MutableInt;
 
 import java.util.ArrayList;
+import java.util.List;
 import java.util.Objects;
+import java.util.function.Function;
 import java.util.function.Supplier;
 
-public final class ReplayingTask<Return> implements Task<Return> {
+public final class ReplayingTask<Input, Output> implements Task<Input, Output> {
   private final Scoped<Context> rootContext;
-  private final Supplier<Return> task;
+  private final Function<Input, Output> task;
 
-  private final ReplayingReactionContext.Memory memory = new ReplayingReactionContext.Memory(new ArrayList<>(), new MutableInt(0));
-
-  public ReplayingTask(final Scoped<Context> rootContext, final Supplier<Return> task) {
+  public ReplayingTask(final Scoped<Context> rootContext, final Function<Input, Output> task) {
     this.rootContext = Objects.requireNonNull(rootContext);
     this.task = Objects.requireNonNull(task);
   }
 
   @Override
-  public TaskStatus<Return> step(final Scheduler scheduler) {
-    final var handle = new ReplayingTaskHandle();
-    final var context = new ReplayingReactionContext(this.rootContext, this.memory, scheduler, handle);
+  public TaskStatus<Output> step(final Scheduler scheduler, final Input input) {
+    // Use locals to avoid holding a reference on `this` from the thread's lambda.
```

Comment:
In this particular case, there are no new threads going on - so perhaps this comment could be
```suggestion
    // Use locals to avoid holding a reference on `this` from the lambda.
```